File: 3634-minimum-removals-to-balance-array/3634-minimum-removals-to-balance-array.py
class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        res = 1
        l = 0
        nums.sort()
        for r in range(len(nums)):
            while nums[l] * k < nums[r]:
                l += 1
            res = max(res, r - l + 1)
        return len(nums) - res

File: 3634-minimum-removals-to-balance-array/3634-minimum-removals-to-balance-array.cpp
class Solution {
public:
    int minRemoval(vector<int>& nums, int k) {
        int res = 1;
        int n = nums.size();
        int l = 0;
        sort(nums.begin(), nums.end());
        for(int r = 0; r < nums.size(); r++)
        {
            while(nums[l] < (float)((float)nums[r] / k))
            {
                l++;
            }
            res = max(res, r - l + 1);
        }
        return n - res;
    }
};